gems/smithy-client/lib/smithy-client.rb

```diff
@@ -31,7 +31,6 @@
 
 require_relative 'smithy-client/plugins/logging'
 require_relative 'smithy-client/plugins/raise_response_errors'
-require_relative 'smithy-client/plugins/net_http'
```

Comment:
I think it would simplify a few places if we followed an autoload strategy for these instead of not requiring it at all here.  I get not wanting to load a potentially optional plugin, but I think using autoload gives us the best of both (and then means we don't have to worry about requiring it in many other places).

Reply:
Yes I think we should go back and do autoloading of clients in a separate PR.